python/cuml/random_projection/random_projection.pyx

```diff
@@ -472,14 +561,28 @@ class SparseRandomProjection(Base, BaseRandomProjection):
 
     def __init__(self, handle=None, n_components='auto', density='auto',
                  eps=0.1, dense_output=True, random_state=None,
-                 verbose=False):
-        Base.__init__(self, handle, verbose)
-        self.gaussian_method = False
-        self.density = density if density != 'auto' else -1.0
+                 verbose=False, output_type=None):
+
+        Base.__init__(self,
+                      handle=handle,
+                      verbose=verbose,
+                      output_type=output_type)
 
         BaseRandomProjection.__init__(
             self,
+            gaussian_method=False,
+            density=(density if density != 'auto' else -1.0),
```

Comment:
The docstring says that density should be in the (0,1] range... any idea where the -1 comes from?

Reply:
Not sure where -1 comes from. This was pulled directly from the existing code: https://github.com/rapidsai/cuml/pull/2928/files/c56ba5be5fb9fc0bb2637ea2fad6c5f78aa9ccf1#diff-77b0e91caa369a50e8c3aaf207c1f239L478